sources/api/netdog/Cargo.toml

```diff
@@ -1,13 +1,18 @@
 [package]
 name = "netdog"
-version = "0.1.0"
+version = "0.1.1"
 authors = ["Ben Cressey <bcressey@amazon.com>"]
 license = "Apache-2.0 OR MIT"
 edition = "2021"
 publish = false
 # Don't rebuild crate just because of changes to README.
 exclude = ["README.md"]
 
+[features]
+default = ["systemd-networkd"]
+wicked = []
+systemd-networkd = []
```

Comment:
It is not a great idea to have mutually exclusive features. It would be better to have a `wicked` feature, and to use `cfg(wicked)` and `cfg(not(wicked))`, otherwise `--all-features` will not compile.

But even so, this doesn't actually solve the problem of removing conditional compilation. We are still compiling conditionally. I would prefer it if the Rust package contained `bin/netdog` and `bin/netdog-wicked` which can both be compiled at the same time.

Reply:
I went this approach to minimize the risk to the code and utilize standard rust/cargo approach for this. While it is true that generally you want to avoid mutually exclusive features, there is also not a valid case to compile with --all-features as a variant is not going to have BOTH wicked and systemd installed on it. Also the cfg(wicket) cfg(not(feature=wicked)) approach limits us to just these two and means if we add a third we'd have to readdress the feature flags entirely instead of adding a third feature. 

I do think I need to add a compile_error if a user tries to compile this with both feature flags, that is definitely a miss. I do understand where you are coming from on the not removing conditional compilation you are true the cargo crate still has conditional compilation. However, bottlerocket as a whole no longer does here as it compiles both netdog-systemd-networkd and netdog-wicked thus ensuring we build both versions.